File: extract_next.py
import tarfile
import os

tar_path = r'e:\prasannasir\next-15.1.7.tgz'
dest_path = r'e:\prasannasir\node_modules\next'

os.makedirs(dest_path, exist_ok=True)

print("Extracting next-15.1.7.tgz directly via Python...")
with tarfile.open(tar_path, 'r:gz') as tar:
    for member in tar.getmembers():
        if member.name.startswith('package/'):
            rel_path = member.name[8:]
            if rel_path:
                target_file = os.path.join(dest_path, rel_path)
                if member.isdir():
                    os.makedirs(target_file, exist_ok=True)
                else:
                    os.makedirs(os.path.dirname(target_file), exist_ok=True)
                    with tar.extractfile(member) as source, open(target_file, 'wb') as target_out:
                        if source:
                            target_out.write(source.read())

print("EXTRACT SUCCESS:", os.path.exists(os.path.join(dest_path, 'package.json')))

File: extract_next.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const dest = path.join(__dirname, 'node_modules', 'next');

if (fs.existsSync(dest)) {
  try {
    fs.rmSync(dest, { recursive: true, force: true });
  } catch (e) {
    console.log('rmSync warn:', e.message);
  }
}
fs.mkdirSync(dest, { recursive: true });

const tempPkg = path.join(__dirname, 'temp_extract');
if (fs.existsSync(tempPkg)) {
  try {
    fs.rmSync(tempPkg, { recursive: true, force: true });
  } catch (e) {}
}
fs.mkdirSync(tempPkg, { recursive: true });

console.log('Unpacking next-15.1.7.tgz...');
execSync('tar -xzf next-15.1.7.tgz -C temp_extract', { cwd: __dirname });

const pkgFolder = path.join(tempPkg, 'package');
fs.cpSync(pkgFolder, dest, { recursive: true });
try {
  fs.rmSync(tempPkg, { recursive: true, force: true });
} catch (e) {}

console.log('SUCCESS! require-hook exists:', fs.existsSync(path.join(dest, 'dist', 'server', 'require-hook.js')));
